torch/ao/quantization/fx/_model_report/model_report_visualizer.py

```diff
@@ -486,25 +487,26 @@ def _get_plottable_data(self, feature_filter: str, module_fqn_filter: str) -> Tu
             if is_valid_per_channel_plot:
                 # gather the x_data and multiple y_data
                 # calculate the number of channels
-                num_channels: int = max(row[self.CHANNEL_NUM_INDEX] for row in channel_table) + 1
+                num_channels: int = max(row[self.CHANNEL_NUM_INDEX] for row in table) + 1
```

Comment:
I feel ideally we should just split these two cases, it might be easier to maintain

Reply:
i actually originally had it as two different cases in an earlier PR and the feedback i got was to merge the code together as much as possible to reduce redundancy. 

I think the main argument against two different cases is that the method will get long and there will be a good amount of redundant code.

Reply:
I feel it's fine to repeat if you can't actually merge them, in this case, the use of `if is_valid_per_channel_plot` is revealing that they can't be truely merged, is it possible to not use this `if is_valid_per_channel_plot`?

Reply:
i tried without it and couldn't get it to work as desired. I split up into distinct sections